akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -308,6 +308,26 @@ object Sink {
     new Sink(scaladsl.Sink.combine(output1.asScala, output2.asScala, seq: _*)(num => strategy.apply(num)))
   }
 
+  /**
+   * Creates a `Sink` that is materialized as an [[akka.stream.javadsl.SinkQueueWithCancel]].
+   * [[akka.stream.javadsl.SinkQueueWithCancel.pull]] method is pulling element from the stream and returns ``CompletionStage[Option[T]]``.
+   * `CompletionStage` completes when element is available.
+   *
+   * Before calling pull method second time need to ensure that request buffer is not full or wait until some of the previous CompletionStages completes.
+   * Pull returns Failed future with ''IllegalStateException'' if request buffer is full.
+   *
+   * `Sink` will request at most number of elements equal to size of `inputBuffer` from
+   * upstream and then stop back pressure.  You can configure size of input
+   * buffer by using [[Sink.withAttributes]] method.
+   *
+   * For stream completion you need to pull all elements from [[akka.stream.javadsl.SinkQueueWithCancel]] including last None
+   * as completion marker
+   *
+   * @see [[akka.stream.javadsl.SinkQueueWithCancel]]
+   */
+  def queue[T](requestBufferSize: Int): Sink[T, SinkQueueWithCancel[T]] =
```

Comment:
Maybe call it `maxConcurrentPulls` or perhaps `parallelism` like `mapAsync` rather, that it is a buffer size is an internal concern.

Reply:
Fixed

Reply:
Fixed in the Scaladsl but still `requestBufferSize` here.